components/board/genericlinux/board.go

```diff
@@ -254,11 +254,43 @@ func (b *sysfsBoard) AnalogReaderByName(name string) (board.AnalogReader, bool)
 }
 
 func (b *sysfsBoard) DigitalInterruptByName(name string) (board.DigitalInterrupt, bool) {
-	// TODO(RSDK-2345): If the name is numerical and doesn't already exist, create it here anyway.
+	if b.usePeriphGpio {
+		return nil, false // Digital interrupts aren't supported.
+	}
+
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
 	interrupt, ok := b.interrupts[name]
+	if ok {
+		return interrupt.interrupt, true
+	}
+
+	// Otherwise, the name is not something we recognize yet. If it appears to be a GPIO pin, we'll
+	// remove its GPIO capabilities and turn it into a digital interrupt.
+	gpio, ok := b.gpios[name]
 	if !ok {
 		return nil, false
 	}
+	if err := gpio.Close(); err != nil {
+		b.logger.Errorf("Unable to close GPIO pin to use as interrupt: %s", err)
+		return nil, false
+	}
+
+	defaultInterruptConfig := board.DigitalInterruptConfig{
+		Name: name,
+		Pin:  name,
+		Type: "basic",
+	}
+	interrupt, err := createDigitalInterrupt(b.cancelCtx, defaultInterruptConfig, b.gpioMappings,
+		&b.activeBackgroundWorkers)
+	if err != nil {
+		b.logger.Errorf("Unable to create digital interrupt pin on the fly: %s", err)
+		return nil, false
+	}
+
+	delete(b.gpios, name)
+	b.interrupts[name] = interrupt
```

Comment:
[q] just so I'm clear on the intended sequence of things : 
if we can create a digital interrupt, create it and be on our way
if we can't with that interrupt name, check if it is currently a gpio pin
if it is a gpio oin, close it, then create the digital interrupt
also delete its name and then rename it as a digital interrupt.

This is closing okay across server restarts?

Reply:
A tweak to the very first line: 
if the digital interrupt _already exists,_ return it and be on our way. 
If it doesn't already exist, check if the requested name is the name of a current GPIO pin. 
If so, close the GPIO pin and create the interrupt pin, remove it from our map of GPIO pins, and add it to the map of interrupt pins.

> This is closing okay across server restarts?

When the server process exits, all of its resources are cleaned up by the OS, including any files (pins) that were opened. If you then start a new server process, it is welcome to open these same pins over again (though they won't re-open until something tries using them again. but that should be fine because the new process needs to build structs for all the components again). 

If you meant to ask about what happens when you change your config and we build a new board struct, that _doesn't_ work right now, but is discussed in RSDK-2683.